### PR TITLE
[UnitTest] Fix "[UnitTest] AttributeError: module 'numpy' has no attribute 'ComplexWarning'" in #7766

### LINKED ISSUE #7766: [UnitTest] AttributeError: module 'numpy' has no attribute 'ComplexWarning'
## 🐛 Bug

```
================================================ short test summary info ================================================
FAILED tests/python/common/sampling/test_sampling.py::test_global_uniform_negative_sampling[int32] - AttributeError: module 'numpy' has no attribute 'ComplexWarning'
FAILED tests/python/common/sampling/test_sampling.py::test_global_uniform_negative_sampling[int64] - AttributeError: module 'numpy' has no attribute 'ComplexWarning'
========================== 2 failed, 2872 passed, 60 skipped, 42 warnings in 148.98s (0:02:28) ==========================
```
## To Reproduce

1. `pip install  dgl -f https://data.dgl.ai/wheels/torch-2.4/cu121/repo.html`
2. `PYTHONPATH=tests:$PYTHONPATH python -m pytest tests/python/common/`

```
dgl                           2.4.0+cu121
numpy                         2.1.0
torch                         2.4.0+cu121
torch_geometric               2.5.3
torcheval                     0.0.7
torchmetrics                  1.4.1
```

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by az15240**
## Description

Fix #7766

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes

Commented out a line that will cause `AttributeError: module 'numpy' has no attribute 'ComplexWarning'`.

### CODE REVIEW

File: tests/python/common/sampling/test_sampling.py [hunk 1]
```diff
@@ -1774,7 +1774,7 @@ def contain_edge(g, sg, u, v):
 
 @pytest.mark.parametrize("dtype", ["int32", "int64"])
 def test_global_uniform_negative_sampling(dtype):
-    warnings.simplefilter("ignore", np.ComplexWarning)
+    # warnings.simplefilter("ignore", np.ComplexWarning)
```
Comment (mfbalin) [inline tests/python/common/sampling/test_sampling.py:1777]: Does the following work?
```suggestion
    warnings.simplefilter("ignore", np.exceptions.ComplexWarning)
```

Comment (Rhett-Ying) [reply tests/python/common/sampling/test_sampling.py:1777]: you cannot simply comment out this line. please try with above suggestion.

**Comment by Rhett-Ying** [comment]
re-name the title to be more meaningful.

**Comment by Rhett-Ying** [comment]
does this issue happen since numpy 2.0?

**Comment by az15240** [comment]
It now worked for `np.exceptions.ComplexWarning`.

**Comment by Rhett-Ying** [comment]
can we safely remove that line?

**Comment by az15240** [comment]
> can we safely remove that line?

No, this warning is still raised. I think this is the final version and we can safely close the PR.

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**